TestAccGitLabProjectApprovalRule_basic test failing in master

The TestAccGitLabProjectApprovalRule_basic failed after merging #525 

Action: https://github.com/gitlabhq/terraform-provider-gitlab/runs/1779107460

Error:

```
2021-01-27T19:00:30.9669258Z -----------------------------------------------------
2021-01-27T19:00:30.9678567Z ##[error]    testing.go:745: Error destroying resource! WARNING: Dangling resources
2021-01-27T19:00:30.9691106Z         may exist. The full state and error is shown below.
2021-01-27T19:00:30.9691620Z         
2021-01-27T19:00:30.9692202Z         Error: Check failed: User still exists
2021-01-27T19:00:30.9692654Z         
2021-01-27T19:00:30.9693034Z         State: <no state>
2021-01-27T19:00:30.9694226Z --- FAIL: TestAccGitLabProjectApprovalRule_basic (12.96s)
```

--- a/gitlab/resource_gitlab_user.go
+++ b/gitlab/resource_gitlab_user.go
@@ -4,7 +4,9 @@ import (
 	"fmt"
 	"log"
 	"strconv"
+	"time"
 
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 	gitlab "github.com/xanzy/go-gitlab"
 )
@@ -179,8 +181,28 @@ func resourceGitlabUserDelete(d *schema.ResourceData, meta interface{}) error {
 
 	id, _ := strconv.Atoi(d.Id())
 
-	_, err := client.Users.DeleteUser(id)
-	// Ignoring error due to some bug in library
-	log.Printf("[DEBUG] Delete gitlab user %s", err)
+	if _, err := client.Users.DeleteUser(id); err != nil {
+		return err
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Timeout: 5 * time.Minute,
+		Target:  []string{"Deleted"},
+		Refresh: func() (interface{}, string, error) {
+			user, resp, err := client.Users.GetUser(id)
+			if resp != nil && resp.StatusCode == 404 {
+				return user, "Deleted", nil
+			}
+			if err != nil {
+				return user, "Error", err
+			}
+			return user, "Deleting", nil
+		},
+	}
+
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Could not finish deleting user %d: %w", id, err)
+	}
+
 	return nil
 }

--- a/gitlab/resource_gitlab_user_test.go
+++ b/gitlab/resource_gitlab_user_test.go
@@ -19,7 +19,7 @@ func TestAccGitlabUser_basic(t *testing.T) {
 	resource.Test(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
-		CheckDestroy: testAccCheckGitlabGroupDestroy,
+		CheckDestroy: testAccCheckGitlabUserDestroy,
 		Steps: []resource.TestStep{
 			// Create a user
 			{

